Tests/SwiftSyntaxMacrosTest/MacroSystemTests.swift

```diff
@@ -1236,4 +1313,44 @@ final class MacroSystemTests: XCTestCase {
       indentationWidth: indentationWidth
     )
   }
+
+  func testPreambleExpansion() {
+    assertMacroExpansion(
+      """
+      @Traced
+      func doSomethingDangerous(operation: String) {
+        print("This is the operation: \\(operation)")
+      }
+      """,
+      expandedSource: """
+
+        func doSomethingDangerous(operation: String) {
+          log("Entering doSomethingDangerous(operation: \\(operation))")
+          defer {
+            log("Exiting doSomethingDangerous(operation:)")
+          }
+            print("This is the operation: \\(operation)")
```

Comment:
Do you know why BasicFormat indents this `print` by four spaces here but everything else with two spaces?

Reply:
I think it wasn't getting formatted at the end. I've reworked this and formatting is more reasonable now.